From 5113e3925f2e893b5895350d17eca2164fe2e10d Mon Sep 17 00:00:00 2001
From: Georgy Kirichenko <kirichenkoga@gmail.com>
Date: Sun, 24 Dec 2023 22:15:44 +0300
Subject: [PATCH] Incorporate protobuf stub instead

In further version of protobuf stubs are marked as final in sake of
optimization.
---
 common/iproto_controlplane.h | 18 ++++++------------
 1 file changed, 6 insertions(+), 12 deletions(-)

diff --git a/common/iproto_controlplane.h b/common/iproto_controlplane.h
index 909a03f..2324629 100644
--- a/common/iproto_controlplane.h
+++ b/common/iproto_controlplane.h
@@ -8,26 +8,20 @@
 
 namespace interface
 {
-class protoControlPlane : protected common::icp_proto::BalancerService::Stub
+class protoControlPlane
 {
 public:
 	protoControlPlane(common::proto::UnixProtobufRpcChannel* channel = new common::proto::UnixProtobufRpcChannel(common::icp_proto::socketPath)) :
-	        common::icp_proto::BalancerService::Stub(channel),
-	        channel(channel)
+	        stub(channel)
 	{
 	}
 
-	~protoControlPlane() override
-	{
-		delete channel;
-	}
-
 	auto balancer_real_flush()
 	{
 		common::icp_proto::Empty request;
 		common::proto::RpcController ctl;
 		common::icp_proto::Empty response;
-		RealFlush(&ctl, &request, &response, nullptr);
+		stub.RealFlush(&ctl, &request, &response, nullptr);
 		if (ctl.Failed())
 		{
 			throw std::string("rpc error: " + ctl.ErrorText());
@@ -38,7 +32,7 @@ class protoControlPlane : protected common::icp_proto::BalancerService::Stub
 	{
 		common::proto::RpcController ctl;
 		common::icp_proto::BalancerRealFindResponse response;
-		RealFind(&ctl, &request, &response, nullptr);
+		stub.RealFind(&ctl, &request, &response, nullptr);
 		if (ctl.Failed())
 		{
 			throw std::string("rpc error: " + ctl.ErrorText());
@@ -50,7 +44,7 @@ class protoControlPlane : protected common::icp_proto::BalancerService::Stub
 	{
 		common::proto::RpcController ctl;
 		common::icp_proto::Empty response;
-		Real(&ctl, &request, &response, nullptr);
+		stub.Real(&ctl, &request, &response, nullptr);
 		if (ctl.Failed())
 		{
 			throw std::string("rpc error: " + ctl.ErrorText());
@@ -58,6 +52,6 @@ class protoControlPlane : protected common::icp_proto::BalancerService::Stub
 	}
 
 protected:
-	common::proto::UnixProtobufRpcChannel* channel;
+	common::icp_proto::BalancerService::Stub stub;
 };
 }
